internal/provider/config_data_source.go

```diff
@@ -0,0 +1,105 @@
+package provider
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+
+	apkotypes "chainguard.dev/apko/pkg/build/types"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"gopkg.in/yaml.v2"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ datasource.DataSource = &ConfigDataSource{}
+
+func NewConfigDataSource() datasource.DataSource {
+	return &ConfigDataSource{}
+}
+
+// ConfigDataSource defines the data source implementation.
+type ConfigDataSource struct {
+}
+
+// ConfigDataSourceModel describes the data source data model.
+type ConfigDataSourceModel struct {
+	Id     types.String        `tfsdk:"id"`
+	Config types.String        `tfsdk:"config"`
+	Data   *ImageConfiguration `tfsdk:"data"`
+}
+
+type ImageConfiguration struct {
+	// TODO(mattmoor): Add the rest of the fields
+	// from types.ImageConfiguration
+	Archs []types.String `tfsdk:"archs"`
+}
```

Comment:
Is the idea that eventually this type could also be used in `BuildResourceModel`, so that folks can put the whole (or at least parts) of the config directly in HCL?

I could see a provider-wide `repositories` and `keyring` that get inherited by all `apko_build`s for example, or a `variable "packages"` that gets piped into `apko_build.foo.config.packages`.

Reply:
Yeah, I don't want to block on that, but I think there are a lot of fun things we could do in that direction once we unblock attestations.